modules/photo/include/opencv2/ccm.hpp

```diff
@@ -0,0 +1,520 @@
+\f[
+ln(C_{sl})=f(ln(C_s)), \qquad C_s>0\\
+C_{sl}=0, \qquad C_s=0
+\f]
+
+Because \f$exp(ln(0))\to\infty \f$, the channel whose component is 0 is directly mapped to 0 in the formula above.
+
+For fitting channels respectively, we have:
+\f[
+r=polyfit(ln(R_s),ln(R_{dl}))\\
+g=polyfit(ln(G_s),ln(G_{dl}))\\
+b=polyfit(ln(B_s),ln(B_{dl}))\\
+\f]
+Note that the parameter of \f$ln(*) \f$ cannot be 0.
+Therefore, we need to delete the channels whose values are 0 from \f$R_s \f$ and \f$R_{dl} \f$, \f$G_s\f$ and \f$G_{dl}\f$, \f$B_s\f$ and \f$B_{dl}\f$.
+
+Therefore:
+
+\f[
+ln(R_{sl})=r(ln(R_s)), \qquad R_s>0\\
+R_{sl}=0, \qquad R_s=0\\
+ln(G_{sl})=g(ln(G_s)),\qquad G_s>0\\
+G_{sl}=0, \qquad G_s=0\\
+ln(B_{sl})=b(ln(B_s)),\qquad B_s>0\\
+B_{sl}=0, \qquad B_s=0\\
+\f]
+
+For grayscale polynomials, there are also:
+\f[
+f=polyfit(ln(G_{sl}),ln(G_{dl}))
+\f]
+and:
+\f[
+ln(C_{sl})=f(ln(C_s)), \qquad C_s>0\\
+C_sl=0, \qquad C_s=0
+\f]
+*/
+enum LINEAR_TYPE
+{
+
+    LINEARIZATION_IDENTITY,                  ///<no change is made
+    LINEARIZATION_GAMMA,                      ///<gamma correction; Need assign a value to gamma simultaneously
+    LINEARIZATION_COLORPOLYFIT,               ///<polynomial fitting channels respectively; Need assign a value to deg simultaneously
+    LINEARIZATION_COLORLOGPOLYFIT,            ///<logarithmic polynomial fitting channels respectively; Need assign a value to deg simultaneously
+    LINEARIZATION_GRAYPOLYFIT,                ///<grayscale polynomial fitting; Need assign a value to deg and dst_whites simultaneously
+    LINEARIZATION_GRAYLOGPOLYFIT              ///<grayscale Logarithmic polynomial fitting;  Need assign a value to deg and dst_whites simultaneously
+};
+
+/** @brief Enum of possible functions to calculate the distance between colors.
+
+See https://en.wikipedia.org/wiki/Color_difference for details
+*/
+enum DISTANCE_TYPE
+{
+    DISTANCE_CIE76,                      ///<The 1976 formula is the first formula that related a measured color difference to a known set of CIELAB coordinates.
+    DISTANCE_CIE94_GRAPHIC_ARTS,         ///<The 1976 definition was extended to address perceptual non-uniformities.
+    DISTANCE_CIE94_TEXTILES,
+    DISTANCE_CIE2000,
+    DISTANCE_CMC_1TO1,                   ///<In 1984, the Colour Measurement Committee of the Society of Dyers and Colourists defined a difference measure, also based on the L*C*h color model.
+    DISTANCE_CMC_2TO1,
+    DISTANCE_RGB,                        ///<Euclidean distance of rgb color space
+    DISTANCE_RGBL                        ///<Euclidean distance of rgbl color space
+};
+
+/** @brief Core class of ccm model
+
+Produce a ColorCorrectionModel instance for inference
+*/
+class CV_EXPORTS_W ColorCorrectionModel
```

Comment:
I propose to add read/write methods to dump the model and restore it back from FileStorage